tests/common/devices.py

```diff
@@ -1754,6 +1793,90 @@ def _run_on_asics(self, *module_args, **complex_args):
             else:
                 raise ValueError("Argument 'asic_index' must be an int or string 'all'.")
 
+    def get_frontend_asic_ids(self):
+        if self.sonichost.facts['num_asic'] == 1:
+            return [DEFAULT_ASIC_ID]
+
+        frontend_asic_list = []
+        for asic in self.frontend_asics:
+            frontend_asic_list.append(asic.asic_index)
+
+        return frontend_asic_list
```

Comment:
How about simplify lines 1800-1804 it as a single line:
```
return [asic.asic_index for asic in self.frontend_asics]
```

Code in other similar methods can be simplified too.